hashicorp/waypoint
# ui: removed Waypoint's custom IconTile component

### NOTE: adding no changelog label since this is a PR for a feature branch, which we'll make a PR for to merge into main which will then include a changelog for all changes added to that feature branch

### What changed
- Replaced Waypoint's `IconTile` component with design-systems-components `Hds::IconTile`
- fixed onboarding header icons to use `FlightIcon` instead of hacking `IconTile`
- added `ItemList` component to reduce code clutter

**Credits to @amyrlam for lots of the beginning work on IconTile and @sdav9375 for the `ItemList` code**

### Screenshots
<img width="1431" alt="Screen Shot 2022-03-10 at 4 53 59 PM" src="https://user-images.githubusercontent.com/60155296/157761078-e4184620-bbfc-4bd2-a1cf-0260b121f7e8.png">
<img width="1436" alt="Screen Shot 2022-03-10 at 4 54 13 PM" src="https://user-images.githubusercontent.com/60155296/157761079-d2b92d48-5bc6-48f9-b53a-3cca24376548.png">

### How to test
1. Pull down branch
2. `yarn start`
3. Check the projects and applications list pages
4. Check the onboarding pages
